cvmfs/file_watcher.h

```diff
@@ -0,0 +1,75 @@
+/**
+ * This file is part of the CernVM File System.
+ */
+
+#ifndef CVMFS_FILE_WATCHER_H_
+#define CVMFS_FILE_WATCHER_H_
+
+#include <pthread.h>
+
+#include <map>
+#include <string>
+
+namespace file_watcher {
+
+enum Event {
+  kModified,
+  kRenamed,
+  kAttributes,
+  kHardlinked,
+  kDeleted,
+  kInvalid
+};
+
+class EventHandler {
+ public:
+  EventHandler();
+  virtual ~EventHandler();
+
+  /**
+   * Handle function called per event
+   *
+   * @param file_path - the path of the file this event corresponds to
+   * @param event - the type of event
+   * @param clear_handler - (output) set this to true to have the event
+   *                        handler remove from the loop after this call
+   *
+   * Setting the clear_handler parameter to false means that the FileWatcher
+   * object will attempt to re-register the handler for the same file name
+   * in the case a file was delete - useful for continuously watching a file
+   * which may be deleted and recreated
+   */
+  virtual bool Handle(const std::string& file_path,
+                      Event event,
+                      bool* clear_handler) = 0;
```

Comment:
Since this is part of the public API of the feature, I wanted to have it like this:
* return value signals a failure in the handler
* clear_handler output variable is for deciding if the file watcher should clear or keep the handler.

Reply:
I see.  Let's add a comment then on the meaning of the return value.